Sources/Subscription/AccountManager.swift

```diff
@@ -221,19 +230,44 @@ public class AccountManager: AccountManaging {
         }
     }
 
-    public func fetchEntitlements() async -> Result<[Entitlement], Error> {
-        guard let accessToken else { return .failure(EntitlementsError.noAccessToken) }
+    private func fetchRemoteEntitlements() async -> Result<[Entitlement], Error> {
+        guard let accessToken else {
+            subscriptionCache.cleanup()
+            return .failure(EntitlementsError.noAccessToken)
+        }
+
+        let cachedEntitlements: [Entitlement]? = subscriptionCache.object(forKey: Constants.cachedEntitlementsKey)
 
         switch await AuthService.validateToken(accessToken: accessToken) {
-        case .success(let response):
+
+            case .success(let response):
             let entitlements = response.account.entitlements.compactMap { Entitlement(rawValue: $0.product) }
-            return .success(entitlements)
+                if entitlements != cachedEntitlements {
+                    subscriptionCache.set(entitlements, forKey: SubscriptionCache.Component.entitlements.rawValue)
+                    NotificationCenter.default.post(name: .entitlementsUpdated, object: self, userInfo: nil)
+                }
+                return .success(entitlements)
+
+            case .failure(let error):
+                subscriptionCache.cleanup()
```

Comment:
Won’t this accidentally clear the cache when there’s a connection or temporary server error?